.github/workflows/ci.yml

```diff
@@ -124,12 +124,12 @@ jobs:
     - name: tests
       timeout-minutes: 30
       run: |
-        cargo test --workspace --all-features --all-targets
+        cargo test --workspace --all-features --lib --bins --tests --examples
```

Comment:
if we do this change, we still need `cargo check` to make sure things compile on all targets

Reply:
I think this is already done. I only replaced --all-targets with the above for the tests.

```
    - name: check default features
      run: |
        cargo check --workspace --all-targets

    - name: check all features
      run: |
        cargo check --workspace --all-features --all-targets
```